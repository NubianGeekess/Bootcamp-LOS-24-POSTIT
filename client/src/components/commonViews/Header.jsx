import React from 'react'; // eslint-disable-line no-unused-vars
import PropTypes from 'prop-types';
import Logo from '../../images/postit-logo.png';

/**
 * Header Component
 * Displays a page header where required
 *
 * @method Header
 * @returns {Object} JSX
 * @param {Object} props
 */
const Header = ({ children, handleClick, headerClass, container, logoClass }) =>
  <header className={headerClass}>
    <nav className="amber">
      <div className={container}>
        <a href='#/' className={logoClass}>
          <img src={Logo} alt="postit-logo" /></a>
        <a
          aria-label="mobile-menu"
          className="nav-mobile"
          onClick={handleClick}
        >
          <span></span>
          <span></span>
          <span></span>
        </a>
        {children}
      </div>
      <div className="mobile-nav-wrapper">
        <div className="container" id="mobile-nav"></div>
      </div>
    </nav>
  </header>;

Header.defaultProps = {
  handleClick: () => { },
  headerClass: '',
  logoClass: ''
};
Header.propTypes = {
  handleClick: PropTypes.func.isRequired,
  headerClass: PropTypes.string.isRequired,
  logoClass: PropTypes.string.isRequired,
  container: PropTypes.string
};

export default Header;
