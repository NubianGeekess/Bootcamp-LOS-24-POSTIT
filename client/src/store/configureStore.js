import { compose, createStore, applyMiddleware } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import { persistStore, autoRehydrate } from 'redux-persist';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

// Be sure to ONLY add this middleware in development!
const store = process.env.NODE_ENV === 'development' ?
  createStore(
    rootReducer,
    compose(
      applyMiddleware(...[reduxImmutableStateInvariant(), thunk]),
      window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    )
  )
  :
  createStore(
    rootReducer,
    applyMiddleware(thunk),
  );

// persistStore(store);

export default store;
