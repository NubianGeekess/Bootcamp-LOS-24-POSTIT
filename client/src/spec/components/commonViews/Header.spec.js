/* globals expect */
/* eslint-disable no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Header from '../../../components/commonViews/Header';
import mockData from '../../mocks/mockData';

describe('Given Footer component is mounted', () => {
  it('should render self without crashing', () => {
    const tree = shallow(<Header />);
    const wrapper = shallow(<Header />);
    expect(tree.exists()).toBe(true);
    expect(tree.find('header').exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});