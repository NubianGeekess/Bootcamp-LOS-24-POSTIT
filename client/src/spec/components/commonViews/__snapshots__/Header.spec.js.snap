// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Footer component is mounted should render self without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="amber"
      >
        <div
          className={undefined}
        >
          <a
            className={undefined}
            href="#/"
          >
            <img
              alt="postit-logo"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
          </a>
          <a
            aria-label="mobile-menu"
            className="nav-mobile"
            onClick={undefined}
          >
            <span />
            <span />
            <span />
          </a>
        </div>
        <div
          className="mobile-nav"
        >
          <div
            className="container"
            id="mobile-nav"
          />
        </div>
      </nav>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
          >
            <a
              className={undefined}
              href="#/"
            >
              <img
                alt="postit-logo"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </a>
            <a
              aria-label="mobile-menu"
              className="nav-mobile"
              onClick={undefined}
            >
              <span />
              <span />
              <span />
            </a>
          </div>,
          <div
            className="mobile-nav"
          >
            <div
              className="container"
              id="mobile-nav"
            />
          </div>,
        ],
        "className": "amber",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className={undefined}
                href="#/"
              >
                <img
                  alt="postit-logo"
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />
              </a>,
              <a
                aria-label="mobile-menu"
                className="nav-mobile"
                onClick={undefined}
              >
                <span />
                <span />
                <span />
              </a>,
              undefined,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="postit-logo"
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />,
                "className": undefined,
                "href": "#/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "postit-logo",
                  "src": Object {
                    "process": [Function],
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "mobile-menu",
                "children": Array [
                  <span />,
                  <span />,
                  <span />,
                ],
                "className": "nav-mobile",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "a",
            },
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
              id="mobile-nav"
            />,
            "className": "mobile-nav",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "container",
              "id": "mobile-nav",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="amber"
        >
          <div
            className={undefined}
          >
            <a
              className={undefined}
              href="#/"
            >
              <img
                alt="postit-logo"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </a>
            <a
              aria-label="mobile-menu"
              className="nav-mobile"
              onClick={undefined}
            >
              <span />
              <span />
              <span />
            </a>
          </div>
          <div
            className="mobile-nav"
          >
            <div
              className="container"
              id="mobile-nav"
            />
          </div>
        </nav>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className={undefined}
            >
              <a
                className={undefined}
                href="#/"
              >
                <img
                  alt="postit-logo"
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />
              </a>
              <a
                aria-label="mobile-menu"
                className="nav-mobile"
                onClick={undefined}
              >
                <span />
                <span />
                <span />
              </a>
            </div>,
            <div
              className="mobile-nav"
            >
              <div
                className="container"
                id="mobile-nav"
              />
            </div>,
          ],
          "className": "amber",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className={undefined}
                  href="#/"
                >
                  <img
                    alt="postit-logo"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />
                </a>,
                <a
                  aria-label="mobile-menu"
                  className="nav-mobile"
                  onClick={undefined}
                >
                  <span />
                  <span />
                  <span />
                </a>,
                undefined,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="postit-logo"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                  />,
                  "className": undefined,
                  "href": "#/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "postit-logo",
                    "src": Object {
                      "process": [Function],
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "mobile-menu",
                  "children": Array [
                    <span />,
                    <span />,
                    <span />,
                  ],
                  "className": "nav-mobile",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                ],
                "type": "a",
              },
              undefined,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="container"
                id="mobile-nav"
              />,
              "className": "mobile-nav",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "container",
                "id": "mobile-nav",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
